### ISSUE TITLE
[Bug]: the value of `IntrinsicScalarName` is incomplete

### ISSUE THREAD

**Issue by ArcturusZhang** [microsoft/typespec#4042]
### Describe the bug

[The function definition](https://github.com/microsoft/typespec/blob/60d87bc77014a6756b255f3e81f931ea8b25e80f/packages/compiler/src/core/checker.ts#L287) says: if `isStdType` returns true, `type` is a Type and its `name` is one of the `StdTypeName`.
The definition of `StdTypeName` is `IntrinsicScalarName | "Array" | "Record"`.

When I write code and pass in a symbol of `unixTimestamp32` (defined [here](https://github.com/microsoft/typespec/blob/60d87bc77014a6756b255f3e81f931ea8b25e80f/packages/compiler/lib/std/types.tsp#L8)), the `isStdType` returns `true`, but its name is not in the list of `IntrinsicScalarName`.

### Reproduction

Try write a tsp like this:
```typespec
model Test {
  prop: unixTimestamp32;
}
```
and get the type of property `prop`, then put it to `isStdType` and see whether it returns true.

### Checklist

- [X] Follow our [Code of Conduct](https://github.com/microsoft/typespec/blob/main/CODE_OF_CONDUCT.md)
- [X] Check that there isn't already an issue that request the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/Microsoft/typespec/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ArcturusZhang** [comment]
So if `unixTimestamp32` is a std type (it is inside the TypeSpec namespace anyway), we should add this value to `IntrinsicScalarName`.

**Comment by timotheeguerin** [comment]
Because it isn't actually an intrinsic type. It's just an helper type

**Comment by markcowl** [comment]
- add new union for scalars that return true for isStdType and are not in the current union

**Comment by ArcturusZhang** [comment]
> * add new union for scalars that return true for isStdType and are not in the current union

I think this should solve our issue on TCGC side.
Now TCGC's builtin type are built on top of `IntrinsicScalarName`. If we could have a new union for the "std types", we could change our implementation to build on that.

**Comment by ArcturusZhang** [comment]
> Because it isn't actually an intrinsic type. It's just an helper type.
> 
> But maybe what you want is another union that has the std types as well.
> 
> Intrinsic types are the types in intrinsic.tsp that must always be loaded. STD types are the standard library which is loaded by default but could be omitted.

OK - here I am only calling our the inconsistency which I think it is a bug. We could fix it either way, I do not really have a preference on how to fix this.
Whatever way you choose to fix this, please let us know, because TCGC code and type system are built upon `IntrinsicScalarName` right now.

**Comment by ArcturusZhang** [comment]
@timotheeguerin @markcowl does this approach really make sense when the `IntrinsicScalarName` include `url`, but `url` is defined in the same file as `unixTimestamp32`?
I found this fact does not agree with the things you claimed above.

**Comment by timotheeguerin** [comment]
Yeah url shouldn't either but we need a new type std names that include both I think.
